tests/ui/unpretty/mir/global-asm.rs

```diff
@@ -0,0 +1,7 @@
+//@ compile-flags: -Zdump-mir=all --crate-type=lib
```

Comment:
you mentioned `-dump=mir` in one of your comments. That doesn't exist, and `--emit=mir` seemed  to work. So I don't know what you meant exactly, but is this the right dump command? will it not create a bunch of useless files when someone runs the tests suite?

Reply:
`-Zdump-mir=all`. And yes, I think unfortunately your test will end up dumping a bunch of files when the test suite is executed I think 🤔 .

You should probably move this to a `tests/mir-opt` test, and add an annotation for `// EMIT_MIR ...` for just the global asm block, or however you spell that.